### PR TITLE
[NUI] Add UI thread feature

### PR DESCRIPTION
**Pull request by dkdk-ryu**
UI thread means that the events related to UI (e.g. touch, key, window events) are emitted on the UI thread.
In the main thread, the LowBattery, LowMemory and etc events are emitted.

### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI/src/internal/Application/Application.cs [hunk 1]
```diff
@@ -548,6 +598,49 @@ public enum MemoryStatus
         private NUIApplicationAppControlEventCallbackDelegate applicationAppControlEventCallbackDelegate;
         private ApplicationControlSignal appControlSignal;
 
+        // [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/internal/Application/Application.cs:601]: could you remove unused code?
also there are same unused codes shown below, please remove them also.

File: src/Tizen.NUI/src/internal/Application/Application.cs [hunk 2]
```diff
@@ -996,6 +1089,296 @@ private void OnNUIApplicationAppControl(IntPtr application, IntPtr voidp)
             }
         }
 
+
+        /**
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/internal/Application/Application.cs:1093]: could you change the comment style as 
`/// Event for ...`

File: src/Tizen.NUI/src/internal/Application/Application.cs [hunk 3]
```diff
@@ -996,6 +1089,296 @@ private void OnNUIApplicationAppControl(IntPtr application, IntPtr voidp)
...
+            if (applicationTaskTerminateEventHandler != null)
+            {
+                NUIApplicationTerminatingEventArgs e = new NUIApplicationTerminatingEventArgs();
+                e.Application = this;
+                applicationTaskTerminateEventHandler.Invoke(this, e);
+            }
+        }
+
+        /**
+        * @brief Event for LanguageChanged signal which can be used to subscribe/unsubscribe the event handler
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/internal/Application/Application.cs:1175]: same as above.

**Review by dongsug-song — COMMENTED** [review]
could you add any sample code or test case?
I think it would be ok to add sample App under TizenFX/test/ folder to show how this can be used. thank you.

File: src/Tizen.NUI/src/internal/Application/Application.cs [hunk 4]
```diff
@@ -1218,6 +1614,30 @@ public static Application New(int argc, string stylesheet, NUIApplication.Window
...
+            catch (Exception exception)
+            {
+                Tizen.Log.Fatal("NUI", "[Error] got exception during Application New(), this should not occur, message : " + exception.Message);
+                Tizen.Log.Fatal("NUI", "[Error] error line number : " + new StackTrace(exception, true).GetFrame(0).GetFileLineNumber());
+                Tizen.Log.Fatal("NUI", "[Error] Stack Trace : " + exception.StackTrace);
+                throw;
+            }
+
+            ret = new Application(Interop.Application.New(argc, stylesheet, (int)windowMode, Rectangle.getCPtr(positionSize), useUIThread), true);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
```
Comment (pjh9216) [inline src/Tizen.NUI/src/internal/Application/Application.cs:1636]: Why don't you move this into try block?

File: src/Tizen.NUI/src/internal/Application/NUICoreBackend.cs [hunk 5]
```diff
@@ -356,6 +388,7 @@ private void OnAppControl(object source, NUIApplicationAppControlEventArgs e)
             var handler = Handlers[EventType.AppControlReceived] as Action<AppControlReceivedEventArgs>;
             SafeAppControlHandle handle = new SafeAppControlHandle(e.VoidP, false);
             handler?.Invoke(new AppControlReceivedEventArgs(new ReceivedAppControl(handle)));
+            handle.Dispose();
```
Comment (pjh9216) [inline src/Tizen.NUI/src/internal/Application/NUICoreBackend.cs:391]: How about this?

```cs
using SafeAppControlHandle handle = new SafeAppControlHandle(e.VoidP, false);
handler?.Invoke(new AppControlReceivedEventArgs(new ReceivedAppControl(handle)));
```

File: src/Tizen.NUI/src/internal/Application/NUICoreBackend.cs [hunk 6]
```diff
@@ -370,6 +403,119 @@ private void OnPaused(object source, NUIApplicationPausedEventArgs e)
...
+        /// The App control event callback function.
+        /// </summary>
+        /// <param name="source">The application instance.</param>
+        /// <param name="e">The event argument for AppControl.</param>
+        private void OnTaskAppControl(object source, NUIApplicationAppControlEventArgs e)
+        {
+            Log.Info("NUI", "NUICorebackend OnTaskAppControl Called");
+            SafeAppControlHandle handle = new SafeAppControlHandle(e.VoidP, false);
+            coreTask.OnAppControlReceived(new AppControlReceivedEventArgs(new ReceivedAppControl(handle)));
+            handle.Dispose();
```
Comment (pjh9216) [inline src/Tizen.NUI/src/internal/Application/NUICoreBackend.cs:516]: How about this?

```cs
using SafeAppControlHandle handle = new SafeAppControlHandle(e.VoidP, false);
coreTask.OnAppControlReceived(new AppControlReceivedEventArgs(new ReceivedAppControl(handle)));
```

**Review by dongsug-song — APPROVED** [review]
approve

**Review by hjhun — APPROVED** [review]

File: src/Tizen.NUI/src/internal/Application/NUICoreBackend.cs [hunk 7]
```diff
@@ -177,11 +179,26 @@ public void Run(string[] args)
...
+                        application = Application.NewApplication(args, stylesheet, windowMode, windowRectangle);
+                    }
                 }
                 else
                 {
-                    application = Application.NewApplication(args, stylesheet, windowMode);
+                    if (coreTask != null)
+                    {
+                        // The Rectangle(0, 0, 0, 0) means that want to use the full screen size window at 0,0.
+                        application = Application.NewApplication(args, stylesheet, windowMode, new Rectangle(0, 0, 0, 0), true);
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/internal/Application/NUICoreBackend.cs:196]: Could you remove this warning message?
`Call System.IDisposable.Dispose on object created by 'new Rectangle(0, 0, 0, 0)' before all references to it are out of scope.`

For example,
```suggestion
              using (Rectangle rec = new Rectangle(0, 0, 0, 0))
              {
                        application = Application.NewApplication(args, stylesheet, windowMode, rec, true);
```

**Review by Seoyeon2Kim — APPROVED** [review]
LGTM!

**Pull request merged**